src/services/outliningElementsCollector.ts

```diff
@@ -106,6 +107,23 @@ namespace ts.OutliningElementsCollector {
         return regionDelimiterRegExp.exec(lineText);
     }
 
+    function addTopLevelUnattachedCommentSpans(sourceFile: SourceFile, out: Push<OutliningSpan>): void {
+        // Comments which are attached to statements would be included in addNodeOutliningSpans
+        if (sourceFile.statements.length > 0) return;
+
+        // This will instead set up spans for the missing ones
+        forEach(getLeadingCommentRangesOfNode(sourceFile, sourceFile), (range) => {

```

Comment:
I don't know this code at all, but it seems like this must be similar to the code in addNodeOutliningSpans. Maybe enough to share?

Edit:
*Narrator voice* It wasn't.